lib/ansible/modules/cloud/amazon/ec2_vpc_vgw.py

```diff
@@ -202,9 +206,10 @@ def detach_vgw(client, module, vpn_gateway_id, vpc_id=None):
 def create_vgw(client, module):
     params = dict()
     params['Type'] = module.params.get('type')
+    params['AmazonSideAsn'] = module.params.get('asn')
 
     try:
-        response = client.create_vpn_gateway(Type=params['Type'])
+        response = client.create_vpn_gateway(Type=params['Type'], AmazonSideAsn=params['AmazonSideAsn'])
```

Comment:
I think this needs to be `create_vpn_gateway(**params)` so that we don't always pass the AmazonSideAsn when the user doesn't specify it.